<!-- -*- mode: XML -*- -->
<!-- run timoo_pointcloud/CloudNodelet in a nodelet manager -->

<launch>
  <arg name="model" default="" />
  <arg name="calibration" default="" />
  <arg name="manager" default="timoo_nodelet_manager" />
  <arg name="fixed_frame" default="timoo" />
  <arg name="target_frame" default="timoo" />
  <arg name="max_range" default="130.0" />
  <arg name="min_range" default="0.9" />
  <arg name="max_angle" default="360" />
  <arg name="min_angle" default="0" />
  <arg name="organize_cloud" default="false" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)_cloud"
        args="load timoo_pointcloud/CloudNodelet $(arg manager)">
    <param name="model" value="$(arg model)"/>
    <param name="calibration" value="$(arg calibration)"/>
    <param name="fixed_frame" value="$(arg fixed_frame)"/>
    <param name="target_frame" value="$(arg target_frame)"/>
    <param name="max_range" value="$(arg max_range)"/>
    <param name="min_range" value="$(arg min_range)"/>
    <param name="max_angle" value="$(arg max_angle)"/>
    <param name="min_angle" value="$(arg min_angle)"/>
    <param name="organize_cloud" value="$(arg organize_cloud)"/>
  </node>
</launch>
